lib/kang.js

```diff
@@ -80,20 +85,42 @@ function knMakeSource(str)
 		str = str.substr(0, c);
 	}
 
-	c = str.indexOf(':');
-	if (c == -1) {
-		host = str;
+	if ((matched = IPV4_HOST.exec(str)) !== null) {
```

Comment:
I replaced the explicit parsing with regular expressions because I thought that it made it easier to see what sequences of characters were permissible. I can rewrite it to avoid the regular expressions, if you want.